packages/multi-select-combo-box/src/vaadin-multi-select-combo-box-internal.js

```diff
@@ -140,6 +157,39 @@ class MultiSelectComboBoxInternal extends ComboBoxDataProviderMixin(ComboBoxMixi
     this._toggleElement = this.querySelector('.toggle-button');
   }
 
+  /**
+   * Override combo-box method to group selected
+   * items at the top of the overlay.
+   *
+   * @protected
+   * @override
+   */
+  _setDropdownItems(items) {
+    if (this.readonly || !this.groupSelectedItems) {
+      this._dropdownItems = items;
+      return;
+    }
+
+    if (this.topGroup) {
+      const filteredTopItems = this.topGroup.filter((selectedItem) => {
+        return items.some((item) => this._getItemValue(item) === this._getItemValue(selectedItem));
```

Comment:
**nit:** Could we decrease the number of iterations through the items array? Perhaps this could be done by creating a set of selected item values.

Reply:
Updated code to only use one `forEach` and then detect if the item should go to the top group or not.